sql/core/src/main/scala/org/apache/spark/sql/jdbc/TeradataDialect.scala

```diff
@@ -56,8 +56,8 @@ private case class TeradataDialect() extends JdbcDialect with NoLegacyJDBCError
   // scalastyle:off line.size.limit
   // See https://docs.teradata.com/r/Enterprise_IntelliFlex_VMware/SQL-Stored-Procedures-and-Embedded-SQL/SQLSTATE-Mappings/SQLSTATE-Codes
   // scalastyle:on line.size.limit
-  override def isSyntaxErrorBestEffort(exception: SQLException): Boolean = {
-    Option(exception.getSQLState).exists(_.startsWith("42"))
+  override def isSyntaxErrorBestEffort(exception: SQLException): Option[Boolean] = {
+    Some(Option(exception.getSQLState).exists(_.startsWith("42")))
```

Comment:
We can simplify this expression
```suggestion
    Option(exception.getSQLState).map(_.startsWith("42"))
```